src/services/config-variables.service.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+
+// Mapping type for config template variables '$port'
+export type VariableMap = {
+  $port?: string,
+  $projectPath?: string,
+  $projectStarter?: ?string,
+};
+
+// We're using "template" variables inside the project type configuration file (config/project-types.js)
+// so with the following function we can replace the string $port with the real port number e.g. 3000
+// (see type VariableMap for used mapping strings)

```

Comment:
Hm, what's the advantage to doing it this way? We used to just pass the variables along to the function instead of needing to do substitution. Did adding starters increase the complexity here?